static/js/admin/src/IngestFromSourceForm.js

```diff
@@ -0,0 +1,205 @@
+	};
+	
+	IngestFromSourceForm.prototype.constructor = IngestFromSourceForm;
+	IngestFromSourceForm.prototype = Object.create( AbstractFileUploadForm.prototype );
+	
+	IngestFromSourceForm.prototype.open = function(pid) {
+		var self = this;
+		
+		this.dialog = $("<div class='containingDialog'></div>");
+		this.dialog.dialog({
+			autoOpen: true,
+			width: 560,
+			height: 500,
+			modal: true,
+			title: "Ingest from file server"
+		});
+		
+		this.pid = pid;
+		
+		this.loadCandidateList(pid);
+	};
+	
+	IngestFromSourceForm.prototype.loadCandidateList = function() {
+		var self = this;
+		
+		var loadingOverlay = new ModalLoadingOverlay(this.dialog, {
+			autoOpen : false,
+			type : 'icon',
+			dialog : self.dialog
+		});
+		loadingOverlay.open();
+		
+		$.ajax({
+			url : "listSources/" + this.pid,
+			type : "GET"
+		}).done(function(data){
+			loadingOverlay.remove();
+			
+			self.renderCandidateList(data.sources, data.candidates);
+		});
+	};
+	
+	IngestFromSourceForm.prototype.renderCandidateList = function(sources, candidates) {
+		var self = this;
+		
+		// Generate index of strings to search per candidate and reformat candidate fields for display
+		var candidateIndex = [];
+		for (var i = 0; i < candidates.length; i++) {
+			var candidate = candidates[i];
+			candidateIndex.push(candidate.base + candidate.patternMatched);
+			if (candidate.packagingType == "http://purl.org/net/sword/package/BagIt") {
+				candidate.type = "BagIt";
+			}
+			candidate.sizeFormatted = StringUtilities.readableFileSize(candidate.size);
+			var lastSlash = candidate.patternMatched.lastIndexOf("/");
+			candidate.filename = lastSlash == -1? candidate.patternMatched : candidate.patternMatched.substring(lastSlash + 1);
+		}
+		
+		// Build a lookup map of sources
+		var sourceMap = {};
+		for (var i = 0; i < sources.length; i++) {
+			sourceMap[sources[i].id] = sources[i];
+		}
+		
+		var candidatesForm = candidatesTemplate({sources : sourceMap, candidates : candidates});
+		
+		this.dialog.html(candidatesForm);
+		
+		this.updateCandidateSubmitButton();
+		
+		// Bind selection events for result entries
+		this.dialog.on("click", "#ingest_source_candidates .file_browse_entry", function(e){
+			var $this = $(this);
+			if (e.metaKey || e.ctrlKey) {
+				$this.toggleClass("selected");
+			} else {
+				$this.addClass("selected").siblings().removeClass("selected");
+			}
+			
+			self.updateCandidateSubmitButton();
+		});
+		
+		// Filter box functionality
+		this.dialog.find("#candidate_filter").keyup(function() {
+			var $input = $(this);
+			var value = $input.val();
+			
+			// Compute a regular expression for the user's search terms
+			var matchRe = null;
+			if (value && value.trim()) {
+				var pattern = "";
+				var terms = value.trim().split(" ");
+				for (var i = 0; i < terms.length; i++) {
+					var escaped = terms[i].replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
```

Comment:
The regex is a standard replacement regex for escaping regex characters.  The matching is just (*term1*|*term2*) so that users could search for different parts of the path if they can't remember the whole thing.